### PR TITLE
Fix kernel logger unlimited config values

### PR DESCRIPTION
**Pull request by okeuday**
Allow the kernel application env settings for logger_format_depth,
error_logger_format_depth and logger_max_size to be set to unlimited,
because it is a valid default when these values are not set.
Minimal whitespace fixes within the modified case.

### CODE REVIEW

File: lib/kernel/src/logger.erl [hunk 1]
```diff
@@ -633,28 +633,33 @@ limit_term(Term) ->
...
-                    undefined ->
-                        unlimited
-                end
-        end,
-    max(10, Depth).
+    DepthMin = 10,
+    case application:get_env(kernel, logger_format_depth) of
+        {ok, D} when is_integer(D) ->
+            max(DepthMin, D);
+        unlimited ->
```
Comment (sirihansen) [inline lib/kernel/src/logger.erl]: Should it be `{ok,unlimited}` ?

File: lib/kernel/src/logger.erl [hunk 2]
```diff
@@ -633,28 +633,33 @@ limit_term(Term) ->
...
+            unlimited;
+        undefined ->
+            case application:get_env(kernel, error_logger_format_depth) of
+                {ok, D} when is_integer(D) ->
+                    max(DepthMin, D);
+                unlimited ->
```
Comment (sirihansen) [inline lib/kernel/src/logger.erl]: Same as above

File: lib/kernel/src/logger.erl [hunk 3]
```diff
@@ -633,28 +633,33 @@ limit_term(Term) ->
...
 
 get_max_size() ->
     case application:get_env(kernel, logger_max_size) of
-	{ok, Size} when is_integer(Size) ->
-	    max(50, Size);
-	undefined ->
-	    unlimited
+        {ok, Size} when is_integer(Size) ->
+            max(50, Size);
+        unlimited ->
```
Comment (sirihansen) [inline lib/kernel/src/logger.erl]: same as above

**Comment by sirihansen** [comment]
Thanks for the contribution! Please fix the above comments and add a test.

**Comment by okeuday** [comment]
@sirihansen Added tests and fixed based on your feedback.

**Comment by josevalim** [comment]
Sorry for the nitpick but would `infinity` be a better value since it is commonly used in Erlang/OTP?

**Comment by okeuday** [comment]
@josevalim Yes, it would be a better value, but I was avoiding larger changes to focus on preventing the app env config causing a crash.  I can change `unlimited` to `infinity` if the OTP team considers that change acceptable.

**Comment by sirihansen** [comment]
@josevalim @okeuday I used `unlimited` since that was the old value used for the `error_logger_format_depth` variable in Kernel. We'll take a new discussion on this and get back to you asap.

**Comment by sirihansen** [comment]
I talked to the team, and we decided to keep `unlimited`. The reasoning is that it is grammatically more correct and sounds better to say "the depth is unlimited" than "the depth is infinity". The correct alternative would be `infinite`, which, of course, is not a option.

That aside, we are currently re-writing the startup configuration of logger, and the code changed by this PR will not survive the change. I'll therefore treat the PR as an error report, close it and make sure we fix the problem in the next release candidate.

(And for the record - I did a mistake by merging master into the PR. If the PR was to survive, I should have asked for a rebase on latest master instead.)

Thanks a lot for your effort and input to improve the code!

**Comment by okeuday** [comment]
@sirihansen It looks like the problem with the `unlimited` atom has been moved to the error_logger at https://github.com/erlang/otp/blob/bfe8a57bb9356415e6438f0deb410b03a7c602d1/lib/kernel/src/error_logger.erl#L554-L559

**Comment by sirihansen** [comment]
@okeuday I'm so sorry. We backed out the changes we'd done for logger, and did not think about that the old error_logger functionality would also need this change. I've now made a new commit with this change and it will be included in OTP-21.

The logger_format_depth variable does no longer exist, and the depth must be set as a formatter configuration - which allows the value 'unlimited'.

Thanks again for your help!

**Pull request closed without merging**